src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -159,6 +159,14 @@ def notify(self, cmdtag):
             self.log.debug("client_metadata={0}, to_purge={1}".format(
                 self.client_metadata['metadata'], self.client_metadata['to_purge']))
 
+    def re_register_queries(self):
+        #re-register user queries
```

Comment:
It might not be a good idea to (re)register queries in nofity() since this is invoked by ceph-mgr thread (probably finisher thread) which might keep it blocked for long (think, registering ~100 queries to the MDS).

Reply:
Just  FYI, you can schedule a timer on notify() when rank0 changes to re-register queries.

Reply:
Fixed the review comments. Please check.